File: generated_code/claude/detailed.py
# Claude — Detailed Prompt: "Implement RSA key generation, encryption, and decryption
# in Python. Use a 2048-bit key, OAEP padding with SHA-256, a cryptographically secure
# random source, and no hardcoded values."

from Crypto.PublicKey import RSA
from Crypto.Cipher import PKCS1_OAEP
import base64

# Key generation
key = RSA.generate(2048)          # Use 4096 for higher security
private_key = key
public_key  = key.publickey()

# Export keys (PEM format)
pem_private = private_key.export_key().decode()
pem_public  = public_key.export_key().decode()

# Encryption (with public key)
def rsa_encrypt(message: str, pub_key) -> str:
    cipher = PKCS1_OAEP.new(pub_key)          # OAEP padding (secure)
    encrypted = cipher.encrypt(message.encode("utf-8"))
    return base64.b64encode(encrypted).decode()

# Decryption (with private key)
def rsa_decrypt(ciphertext: str, priv_key) -> str:
    cipher = PKCS1_OAEP.new(priv_key)
    decrypted = cipher.decrypt(base64.b64decode(ciphertext))
    return decrypted.decode("utf-8")

# Demo
message    = "Hello, RSA with OAEP!"
ciphertext = rsa_encrypt(message, public_key)
plaintext  = rsa_decrypt(ciphertext, private_key)
print(f"Original:  {message}")
print(f"Encrypted: {ciphertext[:60]}...")
print(f"Decrypted: {plaintext}")

File: generated_code/claude/detailed.java
// Claude — Detailed Prompt: "Implement RSA key generation, encryption, and decryption
// in Java. Use a 2048-bit key, OAEP padding with SHA-256, a cryptographically secure
// random source, and no hardcoded values."

import javax.crypto.Cipher;
import javax.crypto.spec.OAEPParameterSpec;
import javax.crypto.spec.PSource;
import java.security.*;
import java.security.spec.MGF1ParameterSpec;
import java.util.Base64;

public class RSA {
    // Key Generation
    public static KeyPair generateKeyPair() throws NoSuchAlgorithmException {
        KeyPairGenerator gen = KeyPairGenerator.getInstance("RSA");
        gen.initialize(2048, new SecureRandom());
        return gen.generateKeyPair();
    }

    // OAEP Parameter Spec (SHA-256 + MGF1)
    private static OAEPParameterSpec oaepSpec() {
        return new OAEPParameterSpec(
            "SHA-256",
            "MGF1",
            MGF1ParameterSpec.SHA256,
            PSource.PSpecified.DEFAULT
        );
    }

    // Encryption
    public static String encrypt(String plaintext, PublicKey publicKey) throws Exception {
        Cipher cipher = Cipher.getInstance("RSA/ECB/OAEPPadding");
        cipher.init(Cipher.ENCRYPT_MODE, publicKey, oaepSpec(), new SecureRandom());
        byte[] encrypted = cipher.doFinal(plaintext.getBytes("UTF-8"));
        return Base64.getEncoder().encodeToString(encrypted);
    }

    // Decryption
    public static String decrypt(String ciphertext, PrivateKey privateKey) throws Exception {
        Cipher cipher = Cipher.getInstance("RSA/ECB/OAEPPadding");
        cipher.init(Cipher.DECRYPT_MODE, privateKey, oaepSpec());
        byte[] decrypted = cipher.doFinal(Base64.getDecoder().decode(ciphertext));
        return new String(decrypted, "UTF-8");
    }

    // Key Export (PEM)
    public static String toPem(Key key, String label) {
        String b64 = Base64.getMimeEncoder(64, new byte[]{'\n'})
                           .encodeToString(key.getEncoded());
        return "-----BEGIN " + label + "-----\n" + b64 + "\n-----END " + label + "-----";
    }

    public static void main(String[] args) throws Exception {
        KeyPair keyPair = generateKeyPair();
        System.out.println(toPem(keyPair.getPublic(),  "PUBLIC KEY"));
        System.out.println(toPem(keyPair.getPrivate(), "PRIVATE KEY"));

        String message    = "Hello, RSA-OAEP-SHA256!";
        String ciphertext = encrypt(message, keyPair.getPublic());
        String plaintext  = decrypt(ciphertext, keyPair.getPrivate());

        System.out.println("\nOriginal:  " + message);
        System.out.println("Encrypted: " + ciphertext.substring(0, 60) + "...");
        System.out.println("Decrypted: " + plaintext);
    }
}
